File: LeetcodeJava/MaximumOddBinaryNumber.java
class Solution {
    public String maximumOddBinaryNumber(String s) {
        int cnt1 = 0, cnt0 = 0;
        for (char a : s.toCharArray()) {
            if (a == '1') cnt1++;
            else if (a == '0') cnt0++;
        }
        StringBuilder ans = new StringBuilder();
        for (int i = 0; i < cnt1 - 1; i++) {
            ans.append("1");
        }
        for (int i = 0; i < cnt0; i++) {
            ans.append("0");
        }
        ans.append("1");
        return ans.toString();
    }
}

File: LeetcodeCpp/MaximumOddBinaryNumber.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int count1 = 0, count0 = 0;

        for(auto a:s){
            if(a=='1'){
                count1++;
            }
            else if(a=='0'){
                count0++;
            }
        }
        string ans = string(count1-1,'1') + string(count0,'0') + '1';
        return ans;
    }
};